src/test/unit/middlewares/appLanguage.spec.js

```diff
@@ -0,0 +1,67 @@
+const langMiddleware = require('~/middlewares/appLanguage')
+const { INVALID_LANGUAGE } = require('~/consts/errors')
+const {
+  enums: { APP_LANG_ENUM }
+} = require('~/consts/validation')
+const { createError } = require('~/utils/errorsHelper')
+
+jest.mock('~/utils/errorsHelper', () => ({
+  createError: jest.fn((_status, _errorInfo) => {
+    return new Error('test error')
+  })
+}))
+
+describe('langMiddleware', () => {
+  let req
+  let res
+  let next
+
+  beforeEach(() => {
+    req = {
+      acceptsLanguages: jest.fn()
+    }
+    res = {}
+    next = jest.fn()
+    createError.mockClear()
+  })
+
+  it('should set the language if a valid language is provided', () => {
+    req.acceptsLanguages.mockReturnValue(APP_LANG_ENUM[0])
+
+    langMiddleware(req, res, next)
+
+    expect(req.lang).toBe(APP_LANG_ENUM[0])
+    expect(next).toHaveBeenCalled()
+  })
+
+  it('should throw an error if an invalid language is provided', () => {
+    req.acceptsLanguages.mockReturnValue(false)
+    createError.mockImplementation((status, message) => {
+      const error = new Error(message)
+      error.status = status
+      return error
+    })
+    try {
+      langMiddleware(req, res, next)
+    } catch (error) {
+      expect(createError).toHaveBeenCalledWith(400, INVALID_LANGUAGE)
+      expect(next).not.toHaveBeenCalled()
+    }
+  })
+
+  it('should throw an error if no language is provided', () => {
+    req.acceptsLanguages.mockReturnValue(null)
+    createError.mockImplementation((status, message) => {
+      const error = new Error(message)
+      error.status = status
+      return error
+    })
+
+    try {
+      langMiddleware(req, res, next)
+    } catch (error) {
+      expect(createError).toHaveBeenCalledWith(400, INVALID_LANGUAGE)
+      expect(next).not.toHaveBeenCalled()
+    }
```

Comment:
For testing try to delete `throw` Error in langMiddleware and you will see that this test passed. It is because in this case we skip `catch` block.      `expect(createError).toHaveBeenCalledWith(400, INVALID_LANGUAGE)` and  `expect(next).not.toHaveBeenCalled()` should be outside of `catch`. In catch you should compare `error` with expected error.